src/multi_condition_comparisions/_util/__init__.py

```diff
@@ -0,0 +1,3 @@
+from .checks import check_is_integer_matrix, check_is_numeric_matrix
+
+__all__ = ["check_is_integer_matrix", "check_is_numeric_matrix"]
```

Comment:
```suggestion
__all__ = ["is_integer_matrix", "is_numeric_matrix"]
```
nit but I think `check` is unnecessary

Reply:
and other places etc.

Reply:
I kind of like "check" because it doesn't return a boolean but rather raises a ValueError

Reply:
Of a function called `is_integer_matrix` I'd expect a boolean return value